presto-native-execution/presto_cpp/main/LinuxMemoryChecker.cpp

```diff
@@ -49,6 +49,63 @@ class LinuxMemoryChecker : public PeriodicMemoryChecker {
     LOG(INFO) << fmt::format("Changed to using memory stat file {}", statFile_);
   }
 
+  void start() {
+    // Set memMaxFile to "/sys/fs/cgroup/memory/memory.limit_in_bytes" for
+    // cgroup v1 or "sys/fs/cgroup/memory.max" for cgroup v2.
+    struct stat buffer;
```

Comment:
Do we need to re-declare this as a struct here?

Reply:
Tried declaring it as `stat buffer;` and got error:
```
/root/presto/presto-native-execution/./presto_cpp/main/LinuxMemoryChecker.cpp:30:9: error: expected ‘;’ before ‘buffer’
   30 |     stat buffer;
```